model/job-management/src/main/proto/beam_job_api.proto

```diff
@@ -69,12 +69,16 @@ message PrepareJobRequest {
 
 message PrepareJobResponse {
   // (required) The ID used to associate calls made while preparing the job. preparationId is used
-  // to run the job, as well as in other pre-execution APIs such as Artifact staging.
+  // to run the job.
   string preparation_id = 1;
 
   // An endpoint which exposes the Beam Artifact Staging API. Artifacts used by the job should be
   // staged to this endpoint, and will be available during job execution.
   org.apache.beam.model.pipeline.v1.ApiServiceDescriptor artifact_staging_endpoint = 2;
```

Comment:
Sorry, I don't have context of that PR. Does the client_id refers to a common id which is shared across all grpc connection to identify client or is it a service specific id so that the service can identify client connected to it. I think if it is the 1st then we can pack all the relevant info in that id.
However, I like the idea of having explicit token for artifact staging.

Reply:
Sorry. Forgot to link to it: https://github.com/apache/beam/pull/5349#discussion_r190624823.

Reply:
Thanks for the link.
I agree with @lukecwik on adding more information to `ApiServiceDescriptor` as ApiServiceDescriptor is meant to hold all the relevant connection information. In that case it is fair to pack headers for connections in `ApiServiceDescriptor` which the framework should simply pass.
Enhancing `ApiServiceDescriptor` to have headers (for now and credentials etc later) is a more suitable approach.